src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPH = 'alphabet';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField, isReversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  preparedGoods.sort((good, anotherGood) => {
+    switch (sortField) {
+      case SORT_FIELD_LENGTH:
+        return good.length - anotherGood.length;
+      case SORT_FIELD_ALPH:
+        return good.localeCompare(anotherGood);
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const isChanged = () => {
+    return sortField !== '' || isReversed;
+  };
```

Comment:
1. It good practice to add empty line to separete blocks of code
2. Why do we need isChanged functino, can we just use variable?
```suggestion
  const [sortField, setSortField] = useState('');
  const [isReversed, setIsReversed] = useState(false);
  
  const preparedGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
  const isChanged = sortField !== '' || isReversed;
```